src/sentry/lang/java/utils.py

```diff
@@ -1,9 +1,111 @@
+import sentry_sdk
+from symbolic import ProguardMapper
+
+from sentry import features
+from sentry.attachments import CachedAttachment, attachment_cache
+from sentry.eventstore.models import Event
+from sentry.models import Project, ProjectDebugFile
+from sentry.utils import json
+from sentry.utils.cache import cache_key_for_event
 from sentry.utils.safe import get_path
 
+CACHE_TIMEOUT = 3600
+
 
 def has_proguard_file(data):
     """
     Checks whether an event contains a proguard file
     """
     images = get_path(data, "debug_meta", "images", filter=True)
     return get_path(images, 0, "type") == "proguard"
+
+
+def get_proguard_uuid(event: Event):
+    uuid = None
+    if "debug_meta" in event:
+        images = event["debug_meta"].get("images", [])
+        if not isinstance(images, list):
+            return
+        if event.get("project") is None:
+            return
+
+        for image in images:
+            if image.get("type") == "proguard":
+                uuid = image.get("uuid")
+
+    return uuid
+
+
+def get_proguard_mapper(uuid: str, project: Project):
+    with sentry_sdk.start_span(op="proguard.get_proguard_mapper"):
+        dif_paths = ProjectDebugFile.difcache.fetch_difs(project, [uuid], features=["mapping"])
+        debug_file_path = dif_paths.get(uuid)
+        if debug_file_path is None:
+            return
+
+        mapper = ProguardMapper.open(debug_file_path)
+    if not mapper.has_line_info:
+        return
+
+    return mapper
```

Comment:
There's code that is very similar to the two functions above in https://github.com/getsentry/sentry/blob/master/src/sentry/lang/java/plugin.py#L16, any way we could reuse some of that code here, or vice versa?

Reply:
I took a look and I replaced my logic to get the proguard UUIDs with a helper derived from the method in the stack trace processor, but because the logic for processing the stack trace had more involved wrt setting error messages, I decided to leave that part alone for now.